pyogrio/_io.pyx

```diff
@@ -158,10 +158,17 @@ cdef void* ogr_open(const char* path_c, int mode, char** options) except NULL:
         return ogr_dataset
 
     except NullPointerError:
-        raise DataSourceError("Failed to open dataset (mode={}): {}".format(mode, path_c.decode("utf-8"))) from None
+        raise DataSourceError(
+            "Failed to open dataset (mode={}): {}".format(mode, path_c.decode("utf-8"))
+        ) from None
 
     except CPLE_BaseError as exc:
-        raise DataSourceError(str(exc))
+        if str(exc).endswith("not recognized as a supported file format."):
+            raise DataSourceError(
+                f"{str(exc)} It might help to specify the correct driver explicitly by "
+                "prefixing the file path with '<DRIVER>:', e.g. 'GPKG:path'."
```

Comment:
```suggestion
                "prefixing the file path with '<DRIVER>:', e.g. 'CSV:path'."
```

Let's maybe take an example where it matters? (from reading the other comments, it seems that for specifically GPKG this is not documented on the GDAL side and potentially also not needed because GDAL recognizes GPKG regardless of the extension?)

Reply:
True! Changed.